src/components/form.tsx

```diff
@@ -1,204 +1,151 @@
-        consent: (formElements.namedItem('consent') as HTMLInputElement).checked ? 'yes' : '',
-      };
-    }
-
-    return {
-      name: '',
-      surname: '',
-      birthday: '',
-      gender: '',
-      country: '',
-      picture: '',
-      consent: '',
-    };
-  };
-
-  handleSubmit = (e: React.FormEvent) => {
-    e.preventDefault();
-
-    const formValues = this.getFormValues();
-
-    const { isNotValid, isNotValidInputsCount } = validateForm(formValues);
-
-    if (isNotValidInputsCount) {
-      this.notValidInputsCount = isNotValidInputsCount;
-      this.setState({ isNotValid: isNotValid, isSubmitDisabled: true });
-      return;
-    }
-
-    this.props.addPersonCard(formValues);
-    this.formRef.current?.reset();
-    this.setState({ isSubmitDisabled: true, isFirstFilling: true });
+const errMsg: ErrMsg = {
+  name: 'should not be empty',
+  surname: 'should not be empty',
+  birthday: 'you must be at least 12 yo',
+  picture: 'please select picture',
+  consent: 'please check to proceed',
+  country: 'please select country',
+  gender: 'please select your gender',
+};
+
+const countries = ['Italy', 'Norway', 'Germany', 'Spain', 'Sweden', 'Ukraine', 'USA'];
+const genders = ['male', 'female'];
+
+function Form({ addPersonCard }: IFormProps) {
+  const [isSubmitDisabled, setIsSubmitDisabled] = useState(true);
+  const [isFirstFilling, setIsFirstFilling] = useState(true);
+  const {
+    handleSubmit,
+    reset,
+    register,
+    clearErrors,
+    formState: { errors },
+  } = useForm<IFormValues>({ reValidateMode: 'onSubmit' });
+
+  const submitHandler = (data: IFormValues) => {
+    const { name, surname, birthday, country, gender } = data;
+    const picture = data.picture[0].name;
+
+    addPersonCard({
+      name,
+      surname,
+      birthday,
+      country,
+      gender,
+      picture,
+    });
+    reset();
+    setIsFirstFilling(true);
   };
 
-  resetIsNotValid: React.ChangeEventHandler<HTMLInputElement | HTMLSelectElement> = (e) => {
-    this.notValidInputsCount -= 1;
-
-    if (this.notValidInputsCount === 0) {
-      this.setState({ isSubmitDisabled: false });
+  const clearErrMsg: React.ChangeEventHandler<HTMLInputElement | HTMLSelectElement> = (e) => {
+    clearErrors(e.target.name as keyof IFormValues);
+    if (!Object.keys(errors).length) {
+      setIsSubmitDisabled(false);
     }
-    this.setState((prevState) => {
-      const isNotValid = prevState.isNotValid;
-      const key = e.target.name;
-
-      if (Object.hasOwn(isNotValid, key)) {
-        isNotValid[key as keyof NotValidInputs] = false;
-      }
-
-      return { isNotValid };
-    });
   };
 
-  enableSubmit: React.ChangeEventHandler<HTMLInputElement | HTMLSelectElement> = () => {
-    this.setState({ isSubmitDisabled: false, isFirstFilling: false });
+  const enableSubmit: React.ChangeEventHandler<HTMLInputElement | HTMLSelectElement> = () => {
+    setIsFirstFilling(false);
```

Comment:
I did not use this library but I believe it has something for this https://react-hook-form.com/api/useform/unregister#options